docs/topics/multiplatform/multiplatform-run-tests.md

```diff
@@ -1,67 +1,375 @@
+    ```kotlin
+    fun grep(lines: List<String>, pattern: String, action: (String) -> Unit) {
+        val regex = pattern.toRegex()
+        lines.filter(regex::containsMatchIn)
+            .forEach(action)
     }
-}
-```
+    ```
 
-And the following test in `iosTest`:
+This function is designed to resemble the [UNIX grep command](https://en.wikipedia.org/wiki/Grep). Here, the function
+takes lines of text, a pattern used as a regular expression, and a function that is invoked every time a line matches
+the pattern.
 
-```kotlin
-import kotlin.test.Test
-import kotlin.test.assertTrue
+### Add tests for your common code
+
+Let's test the common code. Before you can do this, it's necessary to create a source set for common tests,
+which has the [`kotlin.test`](https://kotlinlang.org/api/latest/kotlin.test/) API library as a dependency.
+
+1. Navigate to the `build.gradle.kts` file in the `shared` folder. You'll see that there's already a source set for
+   testing the common code. Within its declaration, you have a dependency on the `kotlin.test` library:
+
+    ```kotlin
+    val commonTest by getting {
+        dependencies {
+            implementation(kotlin("test"))
+        }
+    }
+    ```
+
+   Each multiplatform project has a `commonTest` source set by default. This is where the common tests are stored.
+
+2. All you need to do is to create a corresponding folder in your project, which must have the same name. When you
+   create a new directory in the `shared/src` directory, the IDE shows a set of standard options:
+
+   ![Creating common test directory](create-common-test-dir.png){width=300}
+
+   In this case, you need a folder called `commonTest` with the `kotlin` subfolder inside.
+
+3. In the `kotlin` folder, create a new `common.example.search` directory and the `Grep.kt` file.
+4. Update your test file with the following unit test:
+
+    ```kotlin
+    import kotlin.test.Test
+    import kotlin.test.assertContains
+    import kotlin.test.assertEquals
+    
+    class GrepTest {
+        companion object {
+            val sampleData = listOf(
+                "123 abc",
+                "abc 123",
+                "123 ABC",
+                "ABC 123"
+            )
+        }
+    
+        @Test
+        fun shouldFindMatches() {
+            val results = mutableListOf<String>()
+            grep(sampleData, "[a-z]+") {
+                results.add(it)
+            }
+    
+            assertEquals(2, results.size)
+            for (result in results) {
+                assertContains(result, "abc")
+            }
+        }
+    }
+    ```
+
+As you can see, imported annotations and assertions are neither platform nor framework-specific.
+When you run this test later, a platform-specific framework will provide the test runner.
+
+### Run tests
+
+You can run the test by executing:
+
+* The `shouldFindMatches()` test function
+* The `GrepTest` test class
+* The file using the context menu in the **Project View**
```

Comment:
All three options are about running the test, first two use the Run gutter icon, one next to the function and the other near the class. The third option is hidden in the context menu of the file. The Project view was selected in the very beginnig of the tutorial, so that should be clear
Please check the wording if it's clearer now